test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLoginTests.java

```diff
@@ -82,6 +94,35 @@ public void customWithUriVars() {
 		assertThat(request.getRequestURI()).isEqualTo("/uri-login/val1/val2");
 	}
 
+	/**
+	 * spring-restdocs uses postprocessors to do its trick. It will work only if these are merged together
+	 * with our request builders. (gh-7572)
+	 * @throws Exception
+	 */
+	@Test
+	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
+
+		RequestBuilder requestBuilder = formLogin()
+				.user("my-user")
+				.password("my-password")
+				.loginProcessingUrl("/my-path");
+
+		RequestBuilder defaultRequestBuilder = MockMvcRequestBuilders.get("/");
```

Comment:
If you assign this to `MockHttpServletRequestBuilder` we don't need to do the `instanceof` or casting to `ConfigurableSmartRequestBuilder` or `Mergeable`